content/6_Advanced/LineContainer.mdx

```diff
@@ -30,7 +30,180 @@ frequency: 1
+	width="707px"
+	height="416px"
+	frameborder="0"
+/>
+
+This is the main principle of the algorithm. However, there are a few special cases we should cover. In the case of parallel half-planes, there are 2 possible scenarios, either two that are parallel with the same direction or different directions. We should approach parallel lines with different directions the same as if we have to increase the size of the bounding box, since there will have to be at least one of the half-planes within the bounding box in between the two since they are sorted by angle. This means that the only case we have to deal with is where there are two half-planes with the same angle. To do this is straightforward: simply keep the leftmost half-plane and erase the rest since the others will eventually become redundant anyway.
+
+So, as an overview of the algorithm:
+* Sort the set of half-planes by angle; it takes $\mathcal{O}(NlogN)$ time.
+* Iterate over the set. For each, perform incrementation and pop either from the front or back of the double-ended queue as necessary; it takes linear time since every half-plane can only be added or removed once.
+* Once this has been completed, the convex polygon that results from the intersection can be obtained by computing the intersection points of adjacent half-planes in the double-ended queue. It will take linear time. This results in our final complexity as $\{O}(NlogN)$. In a special case where the half-planes are pre-sorted, this algorithm could be completed in $\{O}(N)$. However, for the majority of cases, we will be looking at $\{O}(NlogN)$, most of which is coming from the sorting by angle.
+
+##### Implementation
+
+```cpp
+const long double eps = 1e-9, inf = 1e9;
+
+struct Point {
+
+	long double x, y;
+	explicit Point(long double x = 0, long double y = 0) : x(x), y(y) {}
+
+	friend Point operator+(const Point &p, const Point &q) {
+		return Point(p.x + q.x, p.y + q.y);
+	}
+
+	friend Point operator-(const Point &p, const Point &q) {
+		return Point(p.x - q.x, p.y - q.y);
+	}
+
+	friend Point operator*(const Point &p, const long double &k) {
+		return Point(p.x * k, p.y * k);
+	}
+
+	friend long double cross(const Point &p, const Point &q) {
+		return p.x * q.y - p.y * q.x;
+	}
+};
+
+struct Halfplane {
+
+	Point p, pq;
+	long double angle;
+
+	Halfplane() {}
+	Halfplane(const Point &a, const Point &b) : p(a), pq(b - a) {
+		angle = atan2l(pq.y, pq.x);
+	}
+
+	bool out(const Point &r) { return cross(pq, r - p) < -eps; }
+
+	bool operator<(const Halfplane &e) const {
+		if (fabsl(angle - e.angle) < eps) return cross(pq, e.p - p) < 0;
+		return angle < e.angle;
+	}
+
+	bool operator==(const Halfplane &e) const {
+		return fabsl(angle - e.angle) < eps;
+	}
+
+	friend Point inter(const Halfplane &s, const Halfplane &t) {
+		long double alpha = cross((t.p - s.p), t.pq) / cross(s.pq, t.pq);
+		return s.p + (s.pq * alpha);
+	}
+};
+
+// Actual algorithm
+vector<Point> hp_intersect(vector<Halfplane> &H) {
```

Comment:
1. doc comment explaining what this does
2. rename `H` to something more informative or at least `h` bc snake_case